src/backend/distributed/planner/query_pushdown_planning.c

```diff
@@ -1539,18 +1546,12 @@ SubqueryPushdownMultiNodeTree(Query *originalQuery)
 	 * columnList. Columns mentioned in multiProject node and multiExtendedOp
 	 * node are indexed with their respective position in columnList.
 	 */
-	List *targetColumnList = pull_var_clause_default((Node *) targetEntryList);
+	List *targetColumnList = pull_vars_of_level((Node *) targetEntryList, 0);
 	List *havingClauseColumnList = pull_var_clause_default(queryTree->havingQual);
 	List *columnList = list_concat(targetColumnList, havingClauseColumnList);
 
 	/* create a target entry for each unique column */
-	List *subqueryTargetEntryList = CreateSubqueryTargetEntryList(columnList);
-
-	/*
-	 * Update varno/varattno fields of columns in columnList to
-	 * point to corresponding target entry in subquery target entry list.
-	 */
-	UpdateVarMappingsForExtendedOpNode(columnList, subqueryTargetEntryList);
+	List *subqueryTargetEntryList = CreateSubqueryTargetListAndAdjustVars(columnList);
```

Comment:
It achieves the same, but the previous method of checking for Var equality between the target entries and the columnList no longer worked because we modify varlevelsup when copying a Var to the target list, and it was not obvious how to find the right subquery target entry instead. I think we still could, it just seemed obscure and fragile.

Within this function, it's obvious which Var belongs to which target entry, so doing the adjustment here seemed a bit easier to understand, even if it leads to a weird function name.